api4/user.go

```diff
@@ -1131,6 +1151,37 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(ruser.ToJson()))
 }
 
+// checkProviderAttributes returns the empty string if the patch can be applied without
+// overriding attributes set by the user's login provider; otherwise, the name of the offending
+// field is returned.
+func checkProviderAttributes(a app.AppIface, ouser *model.User, patch *model.UserPatch) string {
+	tryingToChange := func(patchValue *string, userValue *string) bool {
+		return patchValue != nil && *patchValue != *userValue
+	}
+	LdapSettings := &a.Config().LdapSettings
+	SamlSettings := &a.Config().SamlSettings
+
+	conflictField := ""
+	if a.Ldap() != nil &&
```

Comment:
We could do the following for more clarity. Personal preference here but it may be more clear.
```suggestion
	if LdapSettings.Enable &&
```

Reply:
I personally prefer to keep `a.Ldap() ~= nil` since it guarantees that the LDAP interface is present.